Lib/test/test_inspect/test_inspect.py

```diff
@@ -1166,16 +1160,20 @@ def f(self):
                 # This is necessary when the test is run multiple times.
                 sys.modules.pop("inspect_actual")
 
-    @unittest.skipIf(
-        support.is_emscripten or support.is_wasi,
-        "socket.accept is broken"
-    )
     def test_nested_class_definition_inside_async_function(self):
-        import asyncio
-        self.addCleanup(asyncio.set_event_loop_policy, None)
-        self.assertSourceEqual(asyncio.run(mod2.func225()), 226, 227)
+        def run(coro):
+            try:
+                coro.send(None)
```

Comment:
Is the coroutine expected here to be completed in one loop iteration otherwise this be a while loop.

Reply:
Yep it all returns immediately